pkg/network/status/status.go

```diff
@@ -186,68 +177,59 @@ func (m *StatusProber) IsReady(ia *v1alpha1.Ingress, gw map[v1alpha1.IngressVisi
+		// Each Pod is probed using the different hosts, protocol and ports until
+		// one of the probing calls succeeds. Then, the Pod is considered ready and all pending work items
+		// scheduled for that pod are cancelled.
+		ctx, cancel := context.WithCancel(ingCtx)
+		podState := &podState{
+			successCount: 0,
+			context:      ctx,
+			cancel:       cancel,
 		}
 
-		for ip, targets := range targetsPerPod {
-			// Each Pod backing a Gateway is probed using the different hosts, protocol and ports until
-			// one of the probing calls succeeds. Then, the Pod is considered ready and all pending work items
-			// scheduled for that pod are cancelled.
-			ctx, cancel := context.WithCancel(ingCtx)
-			podState := &podState{
-				successCount: 0,
-				context:      ctx,
-				cancel:       cancel,
-			}
+		// Save the podState to be able to cancel it in case of Pod deletion
+		func() {
+			m.mu.Lock()
+			defer m.mu.Unlock()
+			m.podStates[ip] = podState
+		}()
 
-			// Save the podState to be able to cancel it in case of Pod deletion
-			func() {
-				m.mu.Lock()
-				defer m.mu.Unlock()
-				m.podStates[ip] = podState
-			}()
-
-			// Update states and cleanup m.podStates when probing is done or cancelled
-			go func(ip string) {
-				<-podState.context.Done()
-				m.updateStates(ingressState, podState)
-
-				m.mu.Lock()
-				defer m.mu.Unlock()
-				// It is critical to check that the current podState is also the one stored in the map
-				// before deleting it because it could have been replaced if a new version of the ingress
-				// has started being probed.
-				if state, ok := m.podStates[ip]; ok && state == podState {
-					delete(m.podStates, ip)
-				}
-			}(ip)
-
-			for _, host := range hosts.List() {
-				for _, target := range targets {
-					workItem := &workItem{
-						ingressState: ingressState,
-						podState:     podState,
-						url:          fmt.Sprintf(target.urlTmpl, host),
-						podIP:        ip,
-						podPort:      target.targetPort,
-					}
-					workItems = append(workItems, workItem)
+		// Update states and cleanup m.podStates when probing is done or cancelled
+		go func(ip string) {
+			<-podState.context.Done()
+			m.updateStates(ingressState, podState)
+
+			m.mu.Lock()
+			defer m.mu.Unlock()
+			// It is critical to check that the current podState is also the one stored in the map
+			// before deleting it because it could have been replaced if a new version of the ingress
+			// has started being probed.
+			if state, ok := m.podStates[ip]; ok && state == podState {
+				delete(m.podStates, ip)
+			}
+		}(ip)
+
+		for port, urls := range targets {
+			for _, url := range urls.List() {
+				workItem := &workItem{
+					ingressState: ingressState,
+					podState:     podState,
+					url:          url,
+					podIP:        ip,
+					podPort:      port,
 				}
+				workItems = append(workItems, workItem)
 			}
 		}
-		ingressState.pendingCount += int32(len(targetsPerPod))
+		ingressState.pendingCount++
```

Comment:
I don't understand this change.

Reply:
It's because the loop changes and we already group by pod in our new refactored version.